tests/unit/test_grizzly/users/test_iothub.py

```diff
@@ -217,3 +218,119 @@ def test_send(self, iot_hub_parent: GrizzlyScenario, mocker: MockerFixture, griz
             status_code=500,
             status_description='Failed: not_relevant',
         )
+
+    @pytest.mark.usefixtures('iot_hub_parent')
+    def test_send_gzip(self, iot_hub_parent: GrizzlyScenario, mocker: MockerFixture, grizzly_fixture: GrizzlyFixture) -> None:
+        assert isinstance(iot_hub_parent.user, IotHubUser)
+        iot_hub_parent.user.on_start()
+        grizzly = grizzly_fixture.grizzly
+
+        remote_variables = {
+            'variables': transform(grizzly, {
+                'AtomicIntegerIncrementer.messageID': 31337,
+                'AtomicDate.now': '',
+                'messageID': 137,
+            }),
+        }
+        iot_hub_parent.user.add_context(remote_variables)
+        request = cast(RequestTask, iot_hub_parent.user._scenario.tasks()[-1])
+        request.endpoint = 'not_relevant'
+
+        upload_blob = mocker.patch('azure.storage.blob._blob_client.BlobClient.upload_blob', autospec=True)
+        notify_blob_upload_status = mocker.patch('azure.iot.device.IoTHubDeviceClient.notify_blob_upload_status', autospec=True)
+
+        request = cast(RequestTask, iot_hub_parent.user._scenario.tasks()[-1])
+
+        gzip_compress = mocker.patch('gzip.compress', autospec=True, return_value = 'this_is_compressed')
+        request.metadata = {}
+        request.metadata['content_type'] = 'application/octet-stream; charset=utf-8'
+        request.metadata['content_encoding'] = 'gzip'
+        iot_hub_parent.user.request(request)
+
+        gzip_compress.assert_called_once()
+        upload_blob.assert_called_once_with(
+            SOME(BlobClient),
```

Comment:
`ANY(BlobClient)`

if you remove `autspec=True` from the mock, the first argument wouldn't be the instance.